src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -28,16 +28,73 @@
  * Contact : chris@qbittorrent.org
  */
 
-#include <QFile>
-#include <QHostAddress>
+#include "filterparserthread.h"
+
 #include <QDataStream>
+#include <QFile>
 #include <QStringList>
 
 #include "base/logger.h"
-#include "filterparserthread.h"
 
 namespace libt = libtorrent;
 
+namespace
+{
+    class DATIPv4Parser {
+    public:
+        bool tryParse(const char* str) {
+            unsigned char number = 0;
+
+            const char* numberStart = str;
+            char* endptr;
+            for (; *str; ++str) {
+                if (*str == '.') {
+                    long int extractedNum = strtol(numberStart, &endptr, 10);
+                    if ((extractedNum >= 0L) && (extractedNum <=255L))
```

Comment:
In this case `errno` is only set if the value goes out of range. Both limits are handled with the existing checks.